classes/class3/homework.md

```diff
@@ -0,0 +1,41 @@
+# Homework 3
+In this homework you will build your first Express application, create some dynamic Handlebars templates, and store and load information in a MongoDB database using Mongoose.
+## Assignment
+Create an Express application that has the following routes:
+* GET `/cats/new` => Creates a new cat record. A cat should have a random age, a name, and a list of colors.
+  * These should be generated upon creation, not hardcoded.
+  * Optional: Display a verification that a new cat was created, perhaps by stating the details of the new cat.
+* GET `/cats` = > Shows a sorted list of cats by age. This should display their names, colors and age.
+  * The display doesn't have to be pretty as long as its clear. Feel free to explore some HTML formatting, but we will cover it more next time.
+* GET `/cats/bycolor/:color` => Shows a sorted list of cats by age that have a specific color, where `:color` is a parameter, such as "orange", or "grey" that specifies the color.
+* GET `/cats/delete/old` => Deletes the record of the oldest cat (send it to a nice farm in the country). The cat should not longer appear on any lists
+  * Optional: Display a verification that a cat was deleted, perhaps by stating which cat was deleted.
+* Deploy this to Heroku (see [Additional Information](#additional-information))
+
+
+## Submission
+Fill out [this survey](http://goo.gl/forms/pzXSFUl10f) to turn in this assignment.
+Also remember to read through the next [class](../class4) and come prepared with an understanding of:
+* What some of the differences are between client side javascript and the Node javascript you've been using
+* What HTML forms look like and what in what method their information is sent to the server
+
+
+## Additional Information
+This homework will require deploying to Heroku. We talked about it briefly in class but now its time to set it up. First you'll need an account which you can create at https://www.heroku.com/.
+
+* To setup your local computer, install the [Heroku Toolbelt](https://toolbelt.heroku.com/debian).
+* Work through the [Node.js guide](https://devcenter.heroku.com/articles/getting-started-with-nodejs#introduction) to practice deploying a sample application.
+* In order to use MongoDB in Heroku we will use a third-party addon called MongoLab. Follow their [setup guide](https://devcenter.heroku.com/articles/mongolab#adding-mongolab) to create a data store. You only need to read until you get your URI which will look something like this:
+```sh
+heroku config | grep MONGOLAB_URI
```

Comment:
We need to cover environment variables. Nobody should be committing their MongoLab credentials to GitHub. http://www.devfactor.net/2014/12/30/2375-amazon-mistake/